tests/unit/checkmatelib/client_test.py

```diff
@@ -70,8 +70,37 @@ def test_it_with_a_bad_json_payload(self, client, response):
         with pytest.raises(CheckmateException):
             client.check_url("http://bad.example.com")
 
+    # pylint: disable=too-many-arguments
+    @pytest.mark.parametrize("prefix", ("http://", ""))
+    @pytest.mark.parametrize("path", ("/path", ""))
+    @pytest.mark.parametrize(
+        "domain,allowed",
+        (
+            # Local things we allow
+            ("localhost", True),
+            ("127.0.0.1", True),
+            ("0177.0000.0000.0001", True),  # 127.0.0.1 in octal
+            ("0.0.0.0", True),
+            # Local things we don't
+            ("192.168.0.1", False),
+            # Not public domains
+            ("my_private_name", False),
+            ("index.php", False),
+            # Malformed URL (confused for IPV6)
+            ("example.com]", False),
```

Comment:
This is a slightly iffy one. If you try to use `urllib.parse` on a domain like this it goes crazy because it thinks it's an IPV6 address and then totally fails to parse it.

In our case we are probably rejecting it because it's not valid to have brackets in your domain name. In any case, the result is the same: `BadURL`